### PR TITLE
Added converter for Binary Input cluster

### PR DESCRIPTION
**Pull request by wsowa**
Signed-off-by: Witold Sowa <witold.sowa@gmail.com>

### CODE REVIEW

**Comment by wsowa** [comment]
As the commit message says... This PR adds support for binary input cluster

**Comment by cdjackson** [comment]
Thanks. Please can you update the readme as well. I'll try and take a look over this in the coming days.

**Comment by wsowa** [comment]
Done. The double underscore in `BINARY_INPUT__BASIC` hurts my eyes but this is unfortunately name of the enum in `com.zsmartsystems.zigbee.zcl.protocol.ZclClusterType`. Don't want to change it there as it will be a breaking change

**Comment by wsowa** [comment]
Do you expect this to land in 2.5.0? Not sure when the release is scheduled for...

**Comment by cdjackson** [comment]
In theory all major changes should already be merged for 2.5, so based on this, no...

However, I would like to get this, and the update relating to the ZSS library update done if I can, but I will need to do this in the coming days if I can to allow some time for the M6 milestone.

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterBinaryInput.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.binding.zigbee.ZigBeeBindingConstants;
+import org.openhab.binding.zigbee.converter.ZigBeeBaseChannelConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+
+public class ZigBeeConverterBinaryInput extends ZigBeeBaseChannelConverter implements ZclAttributeListener {
```
Comment (cdjackson) [inline org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/converter/ZigBeeConverterBinaryInput.java]: Can you please add a JavsDoc header here

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/ZigBeeBindingConstants.java [hunk 2]
```diff
@@ -86,6 +86,10 @@
     public static final String CHANNEL_LABEL_FANCONTROL = "Fan Control";
     public static final ChannelTypeUID CHANNEL_FANCONTROL = new ChannelTypeUID("zigbee:fancontrol");
 
+    public static final String CHANNEL_NAME_BINARY_INPUT = "binary_input";
```
Comment (cdjackson) [inline org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/ZigBeeBindingConstants.java]: Can I suggest we change this to just ```binaryswitch``` to keep it similar naming to other channels where we only have the underscore if it separates the cluster from the attribute (or something along those lines). Other channels such as ```activepower``` and ```batterylevel``` don't have this.

File: org.openhab.binding.zigbee/src/main/resources/ESH-INF/thing/channels.xml [hunk 3]
```diff
@@ -92,6 +92,15 @@
         <state pattern="%.1f %unit%" readOnly="true" />
     </channel-type>
 
+    <!-- Binary input sensor -->
+    <channel-type id="binary_input">
```
Comment (cdjackson) [inline org.openhab.binding.zigbee/src/main/resources/ESH-INF/thing/channels.xml]: As above, can we use ```binaryinput``` for the channel

**Comment by wsowa** [comment]
Done.
I use a custom built device for testing that base on NXP's ZigBee (3.0 or Pro depending on device version) Cluster Library so I haven't updated supported devices as this wouldn't be useful to anyone.

**Pull request merged**